Sources/CacheAdvance/CacheReader.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  Created by Dan Federman on 12/26/19.
+//  Copyright © 2019 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+final class CacheReader {
+
+    // MARK: Initialization
+
+    /// Creates a new instance of the receiver.
+    ///
+    /// - Parameters:
+    ///   - file: The file URL indicating the desired location of the on-disk store. This file should already exist.
+    ///   - overwriteOldMessages: When `true`,  the cache encodes a pointer to the oldest message after the newest message marker.
+    init(forReadingFrom file: URL, overwriteOldMessages: Bool) throws {
+        reader = try FileHandle(forReadingFrom: file)
+        self.overwriteOldMessages = overwriteOldMessages
+    }
+
+    deinit {
+        try? reader.closeHandle()
+    }
+
+    // MARK: Internal
+
+    var offsetInFileOfOldestMessage: UInt64 = 0
+
+    var offsetInFile: UInt64 {
+        reader.offsetInFile
+    }
+
+    /// Returns the next encodable message, seeking to the beginning of the next message.
+    func nextEncodedMessage() throws -> Data? {
```

Comment:
This method (and the remaining code in this file) came from `FileHandleExtensions`. It was modified somewhat in the transition, and should be reviewed again.